core/FObject.js

```diff
@@ -657,15 +657,28 @@ var FObject = {
   /** Create a deep copy of this object. **/
   deepClone: function() {
     var m = {};
+    var transients = {};
     for ( var key in this.instance_ ) {
       var value = this[key];
       if ( value !== undefined ) {
         var prop = this.model_.getProperty(key);
-        if ( prop && prop.deepCloneProperty )
-          m[key] = prop.deepCloneProperty.call(prop, value);
+        if ( prop && prop.deepCloneProperty ) {
+          if ( prop.transient )
```

Comment:
In order to allow the model to pull in imports and then fill in the missing transients it had to be here, but if we mark properties directly as imports it can move out.